raster/r.what/main.c

```diff
@@ -472,50 +515,102 @@ int main(int argc, char *argv[])
             G_debug(1, "%s|%s at col %d, row %d\n", cache[point].east_buf,
                     cache[point].north_buf, cache[point].col, cache[point].row);
 
-            if (flg.cat->answer) {
-                fprintf(stdout, "%d%s", cache[point].cat, fs);
-            }
-            fprintf(stdout, "%s%s%s%s%s", cache[point].east_buf, fs,
-                    cache[point].north_buf, fs, cache[point].lab_buf);
+            if (strcmp(opt.format->answer, "plain") == 0) {
```

Comment:
Initially, I tried to separate the printing based on format into separate functions but that didn't make the code any cleaner because of the large number of arguments needed to be passed and because this is inside a nested loop which made it difficult to structure the code.